common/djangoapps/student/tests/test_models.py

```diff
@@ -713,6 +715,44 @@ def test_not_enrolled_student_is_enrolled(self):
         assert actual_student.is_active is True
         assert actual_cea.user == student
 
+    @ddt.data(
+        (True),
+        (False)
+    )
+    def test_auto_enrollment_if_course_enrollment_ended(self, default_value):
+        """
+        Make sure that invited students who register after the course
+        enrollment has ended are not enrolled if
+        DISABLE_AUTO_ENROLL_AFTER_ENROLLMENT_END is True and are enrolled
+        if DISABLE_AUTO_ENROLL_AFTER_ENROLLMENT_END is False
+        """
+
+        def register_and_enroll_student():
+            student = self._set_up_invited_student(
+                course=self.course,
+                active=False,
+                enrolled=False
+            )
+            student.is_active = True
+            # trigger the post_save callback
+            student.save()
+            return CourseEnrollment.get_enrollment(student, self.course.id)
+
+        # Set enrollment end date to a past date so that enrollment is ended
+        enrollment_end = datetime.datetime.now(pytz.UTC) - datetime.timedelta(days=2)
+        course_overview = CourseOverviewFactory.create(id=self.course.id)
+        course_overview.enrollment_end = enrollment_end
+        course_overview.save()
+
+        if default_value:
+            assert register_and_enroll_student() is not None
+        else:
+            with override_settings(FEATURES=dict(
+                settings.FEATURES,
+                DISABLE_AUTO_ENROLL_AFTER_ENROLLMENT_END=True
+            )):
```

Comment:
```suggestion
            with override_settings(FEATURES={**settings.FEATURES, 'DISABLE_AUTO_ENROLL_AFTER_ENROLLMENT_END': True}):
```

I find this easier to read, but it might be subjective.

Reply:
Done